### PR TITLE
Support Spring Boot autoconfiguration

### LINKED ISSUE #23: Support Spring Boot autoconfiguration
Provide Spring Boot auto-configuration of Spring CredHub beans instead of requiring `@Import(CredHubConfiguration.class)` in client apps.

### PR DESCRIPTION
**Pull request by daniellavoie**
- Added spring-credhub-starter project
- Updated demo app with spring-credhub-starter

Doc update in #26 

Closes #23

### CODE REVIEW

File: spring-credhub-cloud-connector/src/main/java/org/springframework/credhub/cloud/CredHubInterpolationServiceDataPostProcessor.java [hunk 1]
```diff
@@ -42,7 +44,21 @@
...
+
+			if (credHubProperties.getUrl() != null
+					&& !credHubProperties.getUrl().isEmpty()) {
+				credHubOperations = credHubTemplateFactroy.credHubTemplate(
+						credHubProperties,
+						credHubTemplateFactroy.clientHttpRequestFactoryWrapper());
+			}
+			else {
+				logger.log(Level.WARNING,
+						"System property spring.credhub is undefined. CredHubOperations cannot be initialized, disabling processing of service data");
```
Comment (scottfrederick) [inline spring-credhub-cloud-connector/src/main/java/org/springframework/credhub/cloud/CredHubInterpolationServiceDataPostProcessor.java]: This message doesn't match the property retrieved above (`credhub.url`). Should both of these be `spring.credhub.url`?

File: spring-credhub-starter/src/main/java/org/springframework/credhub/autoconfig/CredHubAutoConfiguration.java [hunk 2]
```diff
@@ -14,56 +14,44 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.credhub.configuration.CredHubTemplateFactory;
 import org.springframework.credhub.core.CredHubProperties;
 import org.springframework.credhub.core.CredHubTemplate;
-import org.springframework.credhub.support.ClientOptions;
 import org.springframework.http.client.ClientHttpRequestFactory;
 
 /**
```
Comment (scottfrederick) [inline spring-credhub-starter/src/main/java/org/springframework/credhub/autoconfig/CredHubAutoConfiguration.java]: The class should have a description.

File: spring-credhub-starter/src/main/java/org/springframework/credhub/autoconfig/CredHubAutoConfiguration.java [hunk 3]
```diff
@@ -14,56 +14,44 @@
...
+	private final CredHubTemplateFactory credHubTemplateFactory = new CredHubTemplateFactory();
 
-	/**
-	 * Create the {@link CredHubProperties} that contains information about the
-	 * CredHub server.
-	 *
-	 * @return the {@link CredHubProperties} bean
-	 */
 	@Bean
+	@ConfigurationProperties(prefix = "spring.credhub")
```
Comment (scottfrederick) [inline spring-credhub-starter/src/main/java/org/springframework/credhub/autoconfig/CredHubAutoConfiguration.java]: Add method javadoc.

File: spring-credhub-cloud-connector/src/main/java/org/springframework/credhub/cloud/CredHubInterpolationServiceDataPostProcessor.java (hunk 1 above)
Comment (daniellavoie) [reply spring-credhub-cloud-connector/src/main/java/org/springframework/credhub/cloud/CredHubInterpolationServiceDataPostProcessor.java]: Typo error indeed. Should be `spring.credhub.url`.

**Pull request merged**